config/config.go

```diff
@@ -161,6 +166,166 @@ type AuthConfig struct {
 	AppendMetadataHeaders bool          `yaml:"append_metadata_headers,omitempty" json:"append_metadata_headers,omitempty"`
 }
 
+// EnvConfigs contains environment configs or URIs to them
+type EnvConfigs struct {
+	// A list of strings containing environment config URIs
+	// Only the local file system is supported currently, e.g., file://path/to/config.yaml
+	ConfigURIs []string `yaml:"config_uris,omitempty" json:"config_uris,omitempty"`
+
+	// A list of environment configs
+	Inline []EnvironmentConfig `yaml:"inline,omitempty" json:"inline,omitempty"`
+}
+
+// EnvironmentConfig is an Apigee Environment-level config for
+// Envoy Adapter. It contains a list of operations for the adapter to
+// perform request authentication and authorization.
+type EnvironmentConfig struct {
+	// Unique ID of the environment config
+	ID string `yaml:"id" json:"id"`
+
+	// A list of proxy configs
+	ProxyConfigs []ProxyConfig
+}
+
+// ProxyConfig has the proxy configuration
+type ProxyConfig struct {
+	// Top-level basepath for the proxy config
+	Basepath string `yaml:"basepath,omitempty" json:"basepath,omitempty"`
+
+	// A list of Operations, names of which must be unique within the proxy config.
+	Operations []APIOperation `yaml:"operations,omitempty" json:"operations,omitempty"`
+}
+
+// An APIOperation associates a set of rules with a set of request matching
```

Comment:
Go readability: Comment should start with the struct name (remove "An ")

Reply:
("An", "A" or "The") is actually allowed by the style decisions. But I realize I am missing a bunch of periods. :)